rest/client.go

```diff
@@ -156,6 +167,26 @@ func (c Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 	return resp, err
 }
 
+// nextFunc knows how to get and parse additional info from uri into v.
+type nextFunc func(v *interface{}, uri string) (*http.Response, error)
```

Comment:
Love it.  This solution feels very Go, like http.HandlerFunc.

Reply:
Oops, resolved this by accident!